app/components/polls/polls-edit/polls-edit-directive.js

```diff
@@ -41,81 +45,49 @@ angular.module('polls.edit.directive', [
         delete window.onbeforeunload;
       });
 
-      var answerUrl = '/cms/api/v1/answer/';
-
       $scope.saveModel = function () {
         if ($scope.model) {
-          $scope.deleteSodaheadAnswers();
 
-          _.forEach($scope.model.answers, function(answer) {
-            if(!$scope.isNew && !answer.notOnSodahead) {
-              $scope.putSodaheadAnswer(answer);
-            }
-            if(!$scope.isNew && answer.notOnSodahead) {
-              $scope.postSodaheadAnswer(answer, $scope.model.$pk);
-            }
-          });
+          if(!$scope.isNew) {
+            Answer.updatePollAnswers($scope);
+            // reset deleted answers
+            $scope.deletedAnswers = [];
+            return Poll.updatePoll($scope.model).then(function (data) {
+              $location.path('/cms/app/polls/edit/' + data.id + '/');
```

Comment:
Do we need to reset the page path in this scenario?  I _think_ the page form should already be up to date with the proper scope?